crates/starknet-devnet/src/main.rs

```diff
@@ -75,11 +76,44 @@ fn print_predeployed_contracts() {
     println!();
 }
 
-fn print_chain_id(chain_id: ChainId) {
+fn log_chain_id(chain_id: ChainId) {
     println!("Chain ID: {} ({})", chain_id, chain_id.to_felt().to_prefixed_hex_str());
 }
 
-pub async fn set_and_log_fork_block(fork_config: &mut ForkConfig) -> Result<(), anyhow::Error> {
+async fn check_forking_spec_version(
+    client: &JsonRpcClient<HttpTransport>,
+) -> Result<(), anyhow::Error> {
+    let origin_spec_version = client.spec_version().await?;
+    if origin_spec_version != RPC_SPEC_VERSION {
+        warn!(
+            "JSON-RPC API version of origin ({}) does not match this Devnet's version ({}).",
+            origin_spec_version, RPC_SPEC_VERSION
+        );
+    }
+    Ok(())
+}
+
+async fn check_forking_chain_id(
+    client: &JsonRpcClient<HttpTransport>,
+    devnet_chain_id: ChainId,
+) -> Result<(), anyhow::Error> {
+    let origin_chain_id = client.chain_id().await?;
+    let devnet_chain_id_felt = devnet_chain_id.into();
+    if origin_chain_id != devnet_chain_id_felt {
+        warn!(
+            "Origin chain ID ({:#x}) does not match this Devnet's chain ID ({:#x}).",
+            origin_chain_id, devnet_chain_id_felt
+        );
+    }
+    Ok(())
+}
+
+/// Logs forking info if forking specified. If block_number not specified, it is set to the latest
```

Comment:
```suggestion
/// Logs forking info if forking specified. If block_number is not specified, it is set to the latest
```

^ total nitpick

Reply:
Well I was aware of this, I thought I was correctly using the programmer jargon, but if it bothers you, I'll change it

Reply:
I fell victim to one of the classic blunders. The change caused the formatting check to fail because the line was too long: https://app.circleci.com/pipelines/github/0xSpaceShard/starknet-devnet-rs/1623/workflows/74a35ffa-8197-4982-90cb-50603114fa19/jobs/2077. Fixed it with a direct commit to main.